src/main/java/com/exactpro/th2/sailfish/utils/CompareFilter.java

```diff
@@ -0,0 +1,193 @@
+        Object tmpValue = null;
+        try {
+            tmpValue = convertValue(value);
+            isNumber = true;
+        } catch (NumberFormatException e) {
+            potentialException = new IllegalArgumentException("Failed to parse value to Number. Value = " + value, e);
+        }
+        if (!isNumber) {
+            try {
+                tmpValue = convertDateValue(value);
+            } catch (DateTimeParseException ex) {
+                ex.addSuppressed(potentialException);
+                throw new IllegalArgumentException("Failed to parse value to Date. Value = " + value, ex);
+            }
+        }
+        this.value = tmpValue;
+        switch (operation) {
+        case MORE:
+            this.stringFormatOperation = ">";
+            break;
+        case LESS:
+            this.stringFormatOperation = "<";
+            break;
+        case NOT_MORE:
+            this.stringFormatOperation = "<=";
+            break;
+        case NOT_LESS:
+            this.stringFormatOperation = ">=";
+            break;
+        default:
+            throw new IllegalArgumentException("Incorrect math operation " + operation);
+        }
+        this.operation = Objects.requireNonNull(operation);
+    }
+
+    @Override
+    public ExpressionResult validate(Object value) throws RuntimeException {
+        Objects.requireNonNull(value);
+        if (!(value instanceof String)) {
+            throw new IllegalArgumentException("Incorrect value type " + value.getClass().getSimpleName());
+        }
+        Comparable<?> tmpValue;
+        if (isNumber) {
+            try {
+                tmpValue = convertValue((String)value);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Failed to parse value to Number. Value = " + value, e);
+            }
+        } else {
+            try {
+                tmpValue = (Comparable<?>)convertDateValue((String)value);
+            } catch (DateTimeParseException ex) {
+                throw new IllegalArgumentException("Failed to parse value to Date. Value = " + value, ex);
+            }
+        }
+
+        return ExpressionResult.create(compareValues(tmpValue, (Comparable<?>)this.value));
+
+    }
+
+    @Override
+    public String getCondition() {
+        return stringFormatOperation + getValue();
+    }
+
+    @Override
+    public String getCondition(Object value) {
+        return value + getCondition() + getValue();
+    }
+
+    @Override
+    public Object getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean hasValue() {
+        return true;
+    }
+
+    private static Comparable<?> convertValue(String value) {
+        if (value.contains(String.valueOf(DecimalFormatSymbols.getInstance().getDecimalSeparator()))) {
+            return new BigDecimal(value);
+        }
+        return Long.parseLong(value);
+    }
+
+    private static Temporal convertDateValue(String value) {
```

Comment:
To avoid a redundant cast
```suggestion
    private static Comparable<?> convertDateValue(String value) {
```